### ISSUE TITLE
KSMBD, LLMNR and WSDD2, may be ?

### ISSUE THREAD

**Issue by CpServiceSpb** [namjaejeon/ksmbd#379]
First of all, thanks for great work.

So, may be it's worst to add to ksmbd LLMNR name resolution (IPv4, IPv6, multicast) and WS-Discovery neighborhood showng/browsing (WSDD2 daemon written at C - https://github.com/Netgear/wsdd2) .
What is your oppinion ?
That it would be possible by conf file values to turn each submodule (LLMNR and/or WSDD) on/off.

**Comment by namjaejeon** [comment]
I don't know about this. I invite related ones.
@neheb do you know about this ?
@Andy2244 do you know about this ?
@BrainSlayer do you know about this ?

**Comment by Andy2244** [comment]
I don't think that is needed, there is also a python based wsd daemon, which should compile fine on any platform that can run python.

I also read some comments about a win10/11 mdns client that "might" support smb announcement, which would remove the need for those wsd daemons?

What i would still like is for ksmbd to use libavahi to directly inject the avahi service xml into a running avahi daemon. Thats how samba does it and its rather trivial to implement, while avoiding a whole lot of maintainer/enduser errors manually managing those avahi service xml files.

**Comment by neheb** [comment]
@Andy2244 I take it umdnsd is not good enough for this?
